### PR TITLE
Inline context creation and custom list for empty annotations

### PR DESCRIPTION
**Pull request by amitjha21**
This PR address issue #493 .
`Compact/brief/inline` context has been created and contextualized in `Chaise`. 
The default display is changed to *MARKDOWN* for newly created context.
Chaise repo PR has been raised for review.

### CODE REVIEW

**Comment by jrchudy** [comment]
Before I review this PR, can you address the white space being changed unnecessarily by your editor? 

@RFSH asked you about this a month ago in a different PR and I asked you about this in your `batch add association` PR in chaise last week. He asked me to make the same change and I have since then so that PR reviews are much easier to go over without having to find what the actual changes are.

**Comment by amitjha21** [comment]
I have changed the editor setting for this one. but it seems not working; I will check this again.

File: js/reference.js [hunk 1]
```diff
@@ -2851,14 +2862,23 @@
...
+                        inbFKItems = true;
                     var values = [];
 
                     // Iterate over all data rows to compute the row values depending on the row_markdown_pattern.
                     for (var i = 0; i < this._data.length; i++) {
-
+                        var value = [];
                         // render template
-                        var value = module._renderTemplate(this._ref.display._markdownPattern, this._data[i], this._ref._table, this._ref._context);
+                        if(inbFKItems)
```
[the remark this answers is not in the document]
Comment (RFSH) [reply js/reference.js]: Bring this statement outside the for loop. The following is also the logic that had to implement.
```javascript
var i, values, pattern;
if (typeof this._ref.display._markdownPattern === 'string') {
   // The previous logic  pattern = ...
} else {
   // the new logic that you should have added
   var values = [];
   for ( i = 0; i < this.tuples.length; i++) {
      var tuple = tuples[i];
      var url = tuple.reference.contextualize.detailed.appLink;
      values.push("* ["+ tuple.displayname.value +"](" + url+ ") " + this._ref.display._separator);
   }
   pattern = this._ref.display._prefix + values.join(" \n") + this._ref.display._suffix;
}

this._content =  module._formatUtils.printMarkdown(pattern);
```

File: js/reference.js [hunk 2]
```diff
@@ -2851,14 +2862,23 @@
                     if (!this._data || !this._data.length) {
                         return null;
                     }
-
+                    var inbFKItems = false;
```
[the remark this answers is not in the document]
Comment (RFSH) [reply js/reference.js]: This is not even needed, look at my other comment that I wrote the implementation.

File: js/reference.js [hunk 3]
```diff
@@ -3097,45 +3110,52 @@
...
+                           if (value !== null) {
+                               values.push(value);
+                           }
+                       }
+
+                       // Join the values array using the separator and prepend it with the prefix and append suffix to it.
+                       pattern = this._ref.display._prefix + values.join(this._ref.display._separator) + this._ref.display._suffix;
+                   }else{
+                       if(this._ref.display._separator === undefined){
+                           this._ref.display._separator = "";
```
Comment (RFSH) [inline js/reference.js]: I forgot about these cases that separator, prefix, and suffix might be undefined. Move these to the `get display` function. Set the initial value for them to the values that are defined there (`\n` for separator, and empty string for other variables).

File: js/reference.js [hunk 4]
```diff
@@ -3097,45 +3110,52 @@
...
+                       }
+                       if(this._ref.display._prefix === undefined){
+                           this._ref.display._prefix = "";
+                       }
+                       if(this._ref.display._suffix === undefined){
+                           this._ref.display._suffix = "";
+                       }
+                      // the new logic that you should have added
```
Comment (RFSH) [inline js/reference.js]: Remove this comment. Also make sure to change the function documentation to include this new logic.

**Pull request merged**